internal/provider/branch_resource.go

```diff
@@ -260,7 +260,7 @@ func (r *branchResource) Create(ctx context.Context, req resource.CreateRequest,
 		Name:         name.ValueString(),
 		ParentBranch: *parentBranch,
 	}
-	if !(data.RestoredFromBranch.IsNull() || data.RestoredFromBranch.IsUnknown()) {
+	if !data.RestoredFromBranch.IsNull() && !data.RestoredFromBranch.IsUnknown() {
```

Comment:
👀 

out of curiosity, what's the lint rule that flagged this?

Reply:
@dctrwatson 
```
internal/provider/branch_resource.go:263:5: QF1001: could apply De Morgan's law (staticcheck)
        if !(data.RestoredFromBranch.IsNull() || data.RestoredFromBranch.IsUnknown()) {
```

Reply:
huh, okay. QF1001 has nothing more than that in their docs too, lol

Also like this tidbit from the wikipedia page (though uncited):
> In modern programming languages, due to the optimisation of compilers and interpreters, the performance differences between these options are negligible or completely absent.